ouroboros-consensus/src/ouroboros-consensus/Ouroboros/Consensus/Storage/ChainDB/Impl/ChainSel.hs

```diff
@@ -519,6 +552,10 @@ chainSelectionForBlock cdb@CDB{..} blockCache hdr punish = do
     -- will first copy the blocks/headers to trim (from the end of the
     -- fragment) from the VolatileDB to the ImmutableDB.
   where
+    loELimit = case cdbLoELimit of
+      LoEDefault   -> k
```

Comment:
Do we need to comment/assert that `k < maxBound`? 😄

Reply:
would this be the only place we'd need to do that? 🙂